worlds/banjo_tooie/__init__.py

```diff
@@ -0,0 +1,1094 @@
+from .WorldOrder import randomize_world_progression
+from BaseClasses import ItemClassification, Tutorial, Item
+from worlds.AutoWorld import World, WebWorld
+from worlds.LauncherComponents import Component, components, Type, launch_subprocess
+
+
+def run_client():
+    from .BTClient import main  # lazy import
+    launch_subprocess(main)
+
+
+components.append(Component("Banjo-Tooie Client", func=run_client, component_type=Type.CLIENT))
+
+
+class BanjoTooieSettings(settings.Group):
+
+    class RomPath(settings.OptionalUserFilePath):
+        """File path of the Banjo-Tooie (USA) ROM."""
+
+    class PatchPath(settings.OptionalUserFolderPath):
+        """Folder path of where to save the patched ROM."""
+
+    class ProgramPath(settings.OptionalUserFilePath):
+        """
+        File path of the program to automatically run.
+        Leave blank to disable.
+        """
+
+    class ProgramArgs(str):
+        """
+        Arguments to pass to the automatically run program.
+        Leave blank to disable.
+        Set to "--lua=" to automatically use the correct path for the lua connector.
+        """
+
+    rom_path: RomPath | str = ""
+    patch_path: PatchPath | str = ""
+    program_path: ProgramPath | str = ""
+    program_args: ProgramArgs | str = "--lua="
+
+
+class BanjoTooieWeb(WebWorld):
+    setup_en = Tutorial(
+        "Setup Banjo-Tooie",
+        """A guide to setting up Archipelago Banjo-Tooie on your computer.""",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Beebaleen"])
+    setup_fr = Tutorial(
+        "Setup Banjo-Tooie",
+        """A guide to setting up Archipelago Banjo-Tooie on your computer.""",
+        "French",
+        "setup_fr.md",
+        "setup/fr",
+        ["g0goTBC"])
+
+    tutorials = [setup_en, setup_fr]
+    option_groups = bt_option_groups
+
+
+class BanjoTooieWorld(World):
+    """
+    Banjo-Tooie is a single-player platform game in which the protagonists are controlled from a third-person perspective.
+    Carrying over most of the mechanics and concepts established in its predecessor,
+    the game features three-dimensional worlds consisting of various platforming challenges and puzzles, with a notable
+    increased focus on puzzle-solving over the worlds of Banjo-Kazooie.
+    """
+
+    game = "Banjo-Tooie"
+    version = "V4.11.3"
+    options: BanjoTooieOptions
+    settings: BanjoTooieSettings
+    settings_key = "banjo_tooie_options"
+    web = BanjoTooieWeb()
+    topology_present = True
+    # item_name_to_id = {name: data.btid for name, data in all_item_table.items()}
+    item_name_to_id = {}
+
+    for name, data in all_item_table.items():
+        if data.btid is None:  # Skip Victory Item
+            continue
+        item_name_to_id[name] = data.btid
+
+    glitches_item_name = itemName.UT_GLITCHED
+    ut_can_gen_without_yaml = True
+
+    location_name_to_id = {name: data.btid for name, data in all_location_table.items()}
```

Comment:
It might be best to not include events in here in case the automatic filter for that is removed, so if it doesn't actually need to be in `all_location_table`, you could remove what I guess is the singular event from there.

Reply:
Added a filter for entries that have a None id.